lib/library_test_utils.go

```diff
@@ -813,485 +798,25 @@ func testCompleteTransaction(t *testing.T) bool {
 	}
 
 	// make sure you panic if transaction complete doesn't return
-	queuedTxCompleted := make(chan struct{}, 1)
-	abortPanic := make(chan struct{}, 1)
-	PanicAfter(10*time.Second, abortPanic, "testCompleteTransaction")
-
-	// replace transaction notification handler
-	var txHash = ""
-	signal.SetDefaultNodeNotificationHandler(func(jsonEvent string) {
-		var envelope signal.Envelope
-		if err := json.Unmarshal([]byte(jsonEvent), &envelope); err != nil {
-			t.Errorf("cannot unmarshal event's JSON: %s. Error %q", jsonEvent, err)
-			return
-		}
-		if envelope.Type == signal.EventSignRequestAdded {
-			event := envelope.Event.(map[string]interface{})
-			t.Logf("transaction queued (will be completed shortly): {id: %s}\n", event["id"].(string))
-
-			completeTxResponse := SignRequestResult{}
-			rawResponse := ApproveSignRequest(C.CString(event["id"].(string)), C.CString(TestConfig.Account1.Password))
-
-			if err := json.Unmarshal([]byte(C.GoString(rawResponse)), &completeTxResponse); err != nil {
-				t.Errorf("cannot decode RecoverAccount response (%s): %v", C.GoString(rawResponse), err)
-			}
-
-			if completeTxResponse.Error != "" {
-				t.Errorf("cannot complete queued transaction[%v]: %v", event["id"], completeTxResponse.Error)
-			}
-
-			txHash = completeTxResponse.Hash
-
-			t.Logf("transaction complete: https://testnet.etherscan.io/tx/%s", txHash)
-			abortPanic <- struct{}{} // so that timeout is aborted
-			queuedTxCompleted <- struct{}{}
-		}
-	})
+	// abortPanic := make(chan struct{}, 1)
+	// PanicAfter(10*time.Second, abortPanic, "testCompleteTransaction")
 
 	// this call blocks, up until Complete Transaction is called
-	txCheckHash, err := statusBackend.SendTransaction(context.TODO(), transactions.SendTxArgs{
+	result := statusBackend.SendTransaction(transactions.SendTxArgs{
 		From:  account.FromAddress(TestConfig.Account1.Address),
 		To:    account.ToAddress(TestConfig.Account2.Address),
 		Value: (*hexutil.Big)(big.NewInt(1000000000000)),
-	})
-	if err != nil {
-		t.Errorf("Failed to SendTransaction: %s", err)
+	}, TestConfig.Account1.Password)
+	if result.Error != nil {
```

Comment:
I would also create a unit test that checks failure scenario.

Reply:
Addressed